servers/services/src/main/java/org/projectnessie/services/rest/TreeResource.java

```diff
@@ -287,14 +303,26 @@ public void mergeRefIntoBranch(String branchName, String hash, Merge merge)
           .merge(
               toHash(merge.getFromRefName(), merge.getFromHash(), true).get(),
               BranchName.of(branchName),
-              toHash(hash, true));
+              toHash(hash, true),
+              getResetMergePropsFunc(MergeOp.MERGE, merge.getFromRefName()));
     } catch (ReferenceNotFoundException e) {
       throw new NessieNotFoundException(e.getMessage(), e);
     } catch (ReferenceConflictException e) {
       throw new NessieConflictException(e.getMessage(), e);
     }
   }
 
+  private BiFunction<Serializer<CommitMeta>, ByteString, ByteString> getResetMergePropsFunc(
```

Comment:
Better move the method into the enum

Reply:
Taking some input parameters, hence the function